### DISCUSSION TITLE
Change progress bar text when completing 100%

### DISCUSSION THREAD

**Discussion by fredericomattos** [Textualize/rich#2353] [Ideas] [answered]
In this example the total progress is 5 seconds (I'm building a model that can be malleable, that I can use any amount of total seconds that will work the same):
```
from rich.progress import Progress
import time
import sys

def main(timeout):
    start_time = time.time()

    penultimate_quarter = timeout-1
    with Progress() as progress:
        task1 = progress.add_task("[green]Processing...", total=100)
        while not progress.finished:
            progress.update(task1, advance=100/penultimate_quarter)
            time.sleep(1)
        time.sleep(1)
        progress.update(task1, description="[blue]Complete Task", advance=100)

    print("--- %s seconds ---" % (time.time() - start_time))

if __name__ == "__main__":
    main(5)
```
Output (same line):  

[![enter image description here][1]][1]

But the output I wanted does not contain this `100%` with the message `Processing...`:  

[![enter image description here][2]][2]

I think it would be interesting to have an option where we can already set the text while it's in progress and set the text when it completes the progress.

  [1]: https://i.stack.imgur.com/1amzc.png
  [2]: https://i.stack.imgur.com/eF4TI.png
  [3]: https://rich.readthedocs.io/en/stable/index.html

**Comment by willmcgugan** [comment, marked as answer]
This would be the expected behaviour. When you advance to 100%, you sleep a total of 2 seconds before setting the description. If you want to change the description text, you should do it as soon as the task is finished.

Try this:

```python
from rich.progress import Progress
import time
import sys

def main(timeout):
    start_time = time.time()

    penultimate_quarter = timeout-1
    with Progress() as progress:
        task1 = progress.add_task("[green]Processing...", total=100)
        time.sleep(1)
        while not progress.finished:
            time.sleep(1)
            progress.update(task1, advance=100/penultimate_quarter)
        progress.update(task1, description="[blue]Complete Task", advance=100)

    print("--- %s seconds ---" % (time.time() - start_time))

if __name__ == "__main__":
    main(5)
```

**Reply by fredericomattos** [reply]
Hi @willmcgugan could you help me with one last detail, please?

If I use the timeout as 5 it runs in 5 seconds, if I use 7 it runs in 7 seconds.

But when we use 31, it runs in 32 seconds.

What is the correct calculation that I need to put so that it always runs perfectly according to the timeout value?
